manager/allocator/cnmallocator/networkallocator.go

```diff
@@ -930,10 +930,29 @@ func initializeDrivers(reg *drvregistry.DrvRegistry) error {
 
 func serviceNetworks(s *api.Service) []*api.NetworkAttachmentConfig {
 	// Always prefer NetworkAttachmentConfig in the TaskSpec
+	networks := s.Spec.Task.Networks
+	filteredNetworks := []*api.NetworkAttachmentConfig{}
 	if len(s.Spec.Task.Networks) == 0 && len(s.Spec.Networks) != 0 {
-		return s.Spec.Networks
+		networks = s.Spec.Networks
 	}
-	return s.Spec.Task.Networks
+	for _, n := range networks {
+		if IsPredefinedNetwork(n.Target) {
+			continue
+		}
+		filteredNetworks = append(filteredNetworks, n)
```

Comment:
As with `taskNetworks`, let's either change callers to explicitly skip predefined networks (there are only two callers), or rename this function to make it clear it's only returning a subset of the networks.